src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -101,6 +122,91 @@ public void resetOnPath(String path) {
         // Intentionally blank
     }
 
+    /**
+     * Helper method which handles the file creation for both normal file uploads and for markdown
+     *
+     * @param file         The file to be uploaded with a maximum file size set in resources/config/application.yml
+     * @param keepFileName specifies if original file name should be kept
+     * @param markdown     boolean which is set to true, when we are uploading a file within the markdown editor
+     * @return The path of the file
+     */
+    @NotNull
+    public String handleSaveFile(MultipartFile file, boolean keepFileName, boolean markdown) {
+        // check for file type
+        String filename = file.getOriginalFilename();
+        if (filename == null) {
+            throw new IllegalArgumentException("Filename cannot be null");
+        }
+        // sanitize the filename and replace all invalid characters with "_"
+        filename = filename.replaceAll("[^a-zA-Z\\d\\.\\-]", "_");
+        String fileExtension = FilenameUtils.getExtension(filename);
+        if (this.allowedFileExtensions.stream().noneMatch(fileExtension::equalsIgnoreCase)) {
+            throw new BadRequestException("Unsupported file type! Allowed file types: " + String.join(", ", this.allowedFileExtensions));
+        }
+
+        final String filePath = markdown ? FilePathService.getMarkdownFilePath() : FilePathService.getTempFilePath();
+        final String fileNameAddition = markdown ? "Markdown_" : "Temp_";
+        final StringBuilder responsePath = new StringBuilder(markdown ? MARKDOWN_FILE_SUBPATH : DEFAULT_FILE_SUBPATH);
+
+        try {
+            File newFile = createNewFile(filePath, filename, fileNameAddition, fileExtension, keepFileName);
+            responsePath.append(newFile.toPath().getFileName());
+
+            // copy contents of uploaded file into newly created file
+            Files.copy(file.getInputStream(), newFile.toPath(), REPLACE_EXISTING);
+
+            return responsePath.toString();
+        }
+        catch (IOException e) {
+            log.error("Could not save file", e);
```

Comment:
```suggestion
            log.error("Could not save file {}", newFile, e);
```
Knowing which file could not be saved might help when debugging via the logs only on a production system.

Reply:
newFile is not available here, I just added the `fileName`.